### PR TITLE
Implement quicksort

### PR DESCRIPTION
**Pull request by rouille**
### Purpose
Implement the quicksort sorting algorithm

### What the code is doing
Select a pivot element (first element of the input array) and partition the array around the pivot into two sub-arrays, one with element less than the pivot and the other with elements greater. The same process is applied recursively on the sub-arrays until the entire array is sorted.

### Testing
New unit tests

### Where to look
A single function, `quicksort`, holds the algorithm

### Usage Example/Visuals
N/A

### Time estimate
10min

### CODE REVIEW

File: interview/algorithm/sort.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+    :param list array: unordered list of numbers/letters.
+    :return: (*list*) -- ordered list of numbers/letters.
+    """
+    if len(array) <= 1:
+        return array
+    else:
+        pivot = array[0]
+        lower = [x for x in array[1:] if x < pivot]
+        greater = [x for x in array[1:] if x > pivot]
```
Comment (BainanXia) [inline interview/algorithm/sort.py]: Should we make it in-place as classic quick sort has memory complexity O(1)?

Comment (rouille) [reply interview/algorithm/sort.py]: Good point

Comment (rouille) [reply interview/algorithm/sort.py]: Done

**Review by BainanXia — APPROVED** [review]
LGTM

**Pull request merged**